### PR TITLE
HPCC-27943 File name in info section of workunits displaying &apos;

### PR DESCRIPTION
**Pull request by kunalaswani**
Corrected to display ' instead of Message with &apos; in field.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by kunalaswani** [comment]
We were displaying messages with the &apos; appended to the front and end of the message in the info section. due to a mismatch.

File: esp/src/src/Utility.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export function xmlEncode2(str) {
     str = "" + str;
     return str.replace(/&/g, "&amp;")
         .replace(/"/g, "&quot;")
-        .replace(/'/g, "&apos;")
+        .replace(/"/g, "&apos;")
```
Comment (jeclrsg) [inline esp/src/src/Utility.ts:23]: This change is incorrect. A single-quote character should be encoded as `&apos;`, a double-quote as `&quot;`.

**Review by jeclrsg — COMMENTED** [review]
@kunalaswani The fix for that issue is probably elsewhere

**Review by jeclrsg — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/InfoGrid.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ export const InfoGrid: React.FunctionComponent<InfoGridProps> = ({
                         }
                         return <><span>{info?.prefix}<Link style={{ marginRight: 3 }} href={`#/workunits/${wuid}/metrics/sg${info.subgraphID}`}>{txt}</Link>{info?.message}</span></>;
                     } else {
```
[the remark this answers is not in the document]
Comment (kunalaswani) [reply esp/src/src-react/components/InfoGrid.tsx]: removal of the else solved the issue all together.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**